### DISCUSSION TITLE
Why was "Create a `public.users` table" removed from the auth docs? Is there a better way to handle extra user data?

### DISCUSSION THREAD

**Discussion by its-danny** [supabase/supabase#2208] [Questions] [answered]
I'm trying to find out the right way to store extra user data. I kept seeing links to [this page](https://supabase.io/docs/guides/auth#create-a-publicusers-table) that doesn't exist anymore. Was able to find a version that still had that section in a [previous version](https://github.com/supabase/supabase/blob/2dcd4bd66143fb64b5848010845f30874abaad01/web/docs/guides/auth.mdx) of the file. But now I'm curious, was it removed because this isn't the correct way to do it? Is there a better approach?

Side question, is it possible to pass extra data for use in that trigger? Storing `full_name` from `new.raw_user_meta_data` in [this example](https://github.com/supabase/supabase/discussions/2175) seems to work because the data is coming from Google OAuth. But what about in the case of a simple email/password sign up form where we also want to store a full name the user enters? Being able to save extra user-entered data on sign up without having to do 2 requests (and dealing w/ situations of one of them not working) would be nice.

If I had to make a suggestion, based on my understanding of how it works atm, you should be able to do:

```ts
await supabase.auth.signUp({
  email,
  password,
  user_metadata: {
    name: 'Johnny'
  }
});
```

That data would end up the same place it does if it's coming from an oauth provider. You can either use it like that, or set up a trigger to move it to a `public.profiles` table when a new user is added to `auth.users`. I'm not sure why email/pass should work different from oauth re extra data. Also because `supabase.auth.update({ data: { name: 'Johnny' } })` puts that data in `auth.users.user_metadata` already.

**Comment by its-danny** [comment, marked as answer]
It looks like the docs for auth have been updated and this has been mostly answered there.

https://supabase.io/docs/guides/auth/managing-user-data

**Reply by kno-raziel** [reply]
Hi @its-danny , did you found a workaround to save extra data for users? I was reading the last link you provided and the suggestion of using a trigger it's seems that will allow only to save in one step the same ID from auth.users into public.users but not any extra data.
